### ISSUE TITLE
[BUG] Can't read 16 bit palette tiff images

### ISSUE THREAD

**Issue by GoodNewsEverybody** [OpenImageIO/oiio#3259]
**Describe the bug**
I'm trying to display an old map with 3d terrain shading, as part of this I use the Import Image As Plane add-on, the image imports as expected, however as soon as I click on the 'Rendered' option for Viewport Shading, Blender crashes, it closes instantly, the original TIFF was only 30MB, I've reduced the resolution down to 5MB and it still crashes the same. I've imported imagery before and have some models with several thousand polygons and no issues... I'm not sure what else to add, please let me know what more info to supply, thanks!

**To Reproduce**
1. Select the Rendered method from the Viewport Shading options after importing a TIFF
2. Then this...
3. Then THIS happens (reproduce the exact error message if you can)

**Expected behavior**
I would expect the image to be displayed as rendered

**Evidence**
Below is the crash log:

# Blender 2.93.4, Commit date: 2021-08-31 09:23, Hash b7205031cec4
bpy.context.space_data.shading.type = 'RENDERED'  # Property

# backtrace
Exception Record:

ExceptionCode         : EXCEPTION_INT_DIVIDE_BY_ZERO
Exception Address     : 0x00007FF61D41A5AD
Exception Module      : blender.exe
Exception Flags       : 0x00000000
Exception Parameters  : 0x0

Stack trace:
blender.exe         :0x00007FF61D41A550  OpenImageIO_v2_1::TIFFInput::palette_to_rgb
blender.exe         :0x00007FF61D41A650  OpenImageIO_v2_1::TIFFInput::read_native_scanline
blender.exe         :0x00007FF61D395F20  OpenImageIO_v2_1::ImageInput::read_native_scanlines
blender.exe         :0x00007FF61D41B330  OpenImageIO_v2_1::TIFFInput::read_native_scanlines
blender.exe         :0x00007FF61D395BE0  OpenImageIO_v2_1::ImageInput::read_native_scanlines
blender.exe         :0x00007FF61D396D20  OpenImageIO_v2_1::ImageInput::read_scanlines
blender.exe         :0x00007FF61D41D280  OpenImageIO_v2_1::TIFFInput::read_scanlines
blender.exe         :0x00007FF61D395240  OpenImageIO_v2_1::ImageInput::read_image
blender.exe         :0x00007FF61D395800  OpenImageIO_v2_1::ImageInput::read_image
blender.exe         :0x00007FF61D395940  OpenImageIO_v2_1::ImageInput::read_image
blender.exe         :0x00007FF61B931770  ccl::oiio_load_pixels<4,unsigned short>
blender.exe         :0x00007FF61B932870  ccl::OIIOImageLoader::load_pixels
blender.exe         :0x00007FF61B82C880  ccl::ImageManager::file_load_image<4,unsigned short>
blender.exe         :0x00007FF61B831B10  ccl::ImageManager::device_load_image
blender.exe         :0x00007FF61B830A90  std::_Func_impl_no_alloc<std::_Binder<std::_Unforced,void (__cdecl ccl::ImageManager::*)(ccl::Devic
blender.exe         :0x00007FF61E1FDFD0  tbb::internal::function_task<std::function<void __cdecl(void)> >::execute
tbb.dll             :0x00007FFE646354D0  tbb::recursive_mutex::scoped_lock::internal_try_acquire
tbb.dll             :0x00007FFE646354D0  tbb::recursive_mutex::scoped_lock::internal_try_acquire
blender.exe         :0x00007FF61A8A2140  tbb::internal::task_group_base::wait
blender.exe         :0x00007FF61E1FE620  ccl::TaskPool::wait_work
blender.exe         :0x00007FF61B8527B0  ccl::GeometryManager::device_update_displacement_images
blender.exe         :0x00007FF61B84E670  ccl::GeometryManager::device_update
blender.exe         :0x00007FF61B8101D0  ccl::Scene::device_update
blender.exe         :0x00007FF61B8131B0  ccl::Scene::update
blender.exe         :0x00007FF61B82AC70  ccl::Session::update_scene
blender.exe         :0x00007FF61B829330  ccl::Session::run_cpu
blender.exe         :0x00007FF61B828F50  ccl::Session::run
blender.exe         :0x00007FF61E200F70  ccl::thread::run
blender.exe         :0x00007FF61B0C9450  std::thread::_Invoke<std::tuple<void * __ptr64 (__cdecl*)(void * __ptr64),ccl::thread * __ptr64>,0,
ucrtbase.dll        :0x00007FFEA56B1B20  configthreadlocale
KERNEL32.DLL        :0x00007FFEA71C7020  BaseThreadInitThunk
ntdll.dll           :0x00007FFEA8342630  RtlUserThreadStart

Threads:
Thread : 00000de4
ntdll.dll           :0x00007FFEA838D3E0  NtDelayExecution
KERNELBASE.dll      :0x00007FFEA51A9590  SleepEx
blender.exe         :0x00007FF61A69EC20  wm_window_process_events
blender.exe         :0x00007FF61A694AA0  WM_main
blender.exe         :0x00007FF61A351C60  main
blender.exe         :0x00007FF61F10C488  __scrt_common_main_seh
KERNEL32.DLL        :0x00007FFEA71C7020  BaseThreadInitThunk
ntdll.dll           :0x00007FFEA8342630  RtlUserThreadStart

Thread : 00002884
ntdll.dll           :0x00007FFEA838CDE0  NtWaitForSingleObject
KERNELBASE.dll      :0x00007FFEA51819D0  WaitForSingleObjectEx
blender.exe         :0x00007FF61E70E410  IlmThread_2_5::Semaphore::wait
blender.exe         :0x00007FF61E70DBE0  IlmThread_2_5::ThreadPool::numThreads
blender.exe         :0x00007FF61E70E880  std::_LaunchPad<std::unique_ptr<std::tuple<void (__cdecl IlmThread_2_5::Thread::*)(void) __ptr64,Il
blender.exe         :0x00007FF61D24CAC0  std::_Pad::_Call_func
ucrtbase.dll        :0x00007FFEA56B1B20  configthreadlocale
KERNEL32.DLL        :0x00007FFEA71C7020  BaseThreadInitThunk
ntdll.dll           :0x00007FFEA8342630  RtlUserThreadStart

Thread : 00003548
ntdll.dll           :0x00007FFEA838CDE0  NtWaitForSingleObject
KERNELBASE.dll      :0x00007FFEA51819D0  WaitForSingleObjectEx
blender.exe         :0x00007FF61E70E410  IlmThread_2_5::Semaphore::wait
blender.exe         :0x00007FF61E70DBE0  IlmThread_2_5::ThreadPool::numThreads
blender.exe         :0x00007FF61E70E880  std::_LaunchPad<std::unique_ptr<std::tuple<void (__cdecl IlmThread_2_5::Thread::*)(void) __ptr64,Il
blender.exe         :0x00007FF61D24CAC0  std::_Pad::_Call_func
ucrtbase.dll        :0x00007FFEA56B1B20  configthreadlocale
KERNEL32.DLL        :0x00007FFEA71C7020  BaseThreadInitThunk
ntdll.dll           :0x00007FFEA8342630  RtlUserThreadStart

Thread : 00002f60
ntdll.dll           :0x00007FFEA838CDE0  NtWaitForSingleObject
KERNELBASE.dll      :0x00007FFEA51819D0  WaitForSingleObjectEx
blender.exe         :0x00007FF61E70E410  IlmThread_2_5::Semaphore::wait
blender.exe         :0x00007FF61E70DBE0  IlmThread_2_5::ThreadPool::numThreads
blender.exe         :0x00007FF61E70E880  std::_LaunchPad<std::unique_ptr<std::tuple<void (__cdecl IlmThread_2_5::Thread::*)(void) __ptr64,Il
blender.exe         :0x00007FF61D24CAC0  std::_Pad::_Call_func
ucrtbase.dll        :0x00007FFEA56B1B20  configthreadlocale
KERNEL32.DLL        :0x00007FFEA71C7020  BaseThreadInitThunk
ntdll.dll           :0x00007FFEA8342630  RtlUserThreadStart

Thread : 00000dcc
ntdll.dll           :0x00007FFEA838CDE0  NtWaitForSingleObject
KERNELBASE.dll      :0x00007FFEA51819D0  WaitForSingleObjectEx
blender.exe         :0x00007FF61E70E410  IlmThread_2_5::Semaphore::wait
blender.exe         :0x00007FF61E70DBE0  IlmThread_2_5::ThreadPool::numThreads
blender.exe         :0x00007FF61E70E880  std::_LaunchPad<std::unique_ptr<std::tuple<void (__cdecl IlmThread_2_5::Thread::*)(void) __ptr64,Il
blender.exe         :0x00007FF61D24CAC0  std::_Pad::_Call_func
ucrtbase.dll        :0x00007FFEA56B1B20  configthreadlocale
KERNEL32.DLL        :0x00007FFEA71C7020  BaseThreadInitThunk
ntdll.dll           :0x00007FFEA8342630  RtlUserThreadStart

Thread : 00002ca8
ntdll.dll           :0x00007FFEA838CDE0  NtWaitForSingleObject
KERNELBASE.dll      :0x00007FFEA51819D0  WaitForSingleObjectEx
blender.exe         :0x00007FF61E70E410  IlmThread_2_5::Semaphore::wait
blender.exe         :0x00007FF61E70DBE0  IlmThread_2_5::ThreadPool::numThreads
blender.exe         :0x00007FF61E70E880  std::_LaunchPad<std::unique_ptr<std::tuple<void (__cdecl IlmThread_2_5::Thread::*)(void) __ptr64,Il
blender.exe         :0x00007FF61D24CAC0  std::_Pad::_Call_func
ucrtbase.dll        :0x00007FFEA56B1B20  configthreadlocale
KERNEL32.DLL        :0x00007FFEA71C7020  BaseThreadInitThunk
ntdll.dll           :0x00007FFEA8342630  RtlUserThreadStart

Thread : 00003538
ntdll.dll           :0x00007FFEA838CDE0  NtWaitForSingleObject
KERNELBASE.dll      :0x00007FFEA51819D0  WaitForSingleObjectEx
blender.exe         :0x00007FF61E70E410  IlmThread_2_5::Semaphore::wait
blender.exe         :0x00007FF61E70DBE0  IlmThread_2_5::ThreadPool::numThreads
blender.exe         :0x00007FF61E70E880  std::_LaunchPad<std::unique_ptr<std::tuple<void (__cdecl IlmThread_2_5::Thread::*)(void) __ptr64,Il
blender.exe         :0x00007FF61D24CAC0  std::_Pad::_Call_func
ucrtbase.dll        :0x00007FFEA56B1B20  configthreadlocale
KERNEL32.DLL        :0x00007FFEA71C7020  BaseThreadInitThunk
ntdll.dll           :0x00007FFEA8342630  RtlUserThreadStart

Thread : 000024c4
ntdll.dll           :0x00007FFEA838CDE0  NtWaitForSingleObject
KERNELBASE.dll      :0x00007FFEA51819D0  WaitForSingleObjectEx
blender.exe         :0x00007FF61E70E410  IlmThread_2_5::Semaphore::wait
blender.exe         :0x00007FF61E70DBE0  IlmThread_2_5::ThreadPool::numThreads
blender.exe         :0x00007FF61E70E880  std::_LaunchPad<std::unique_ptr<std::tuple<void (__cdecl IlmThread_2_5::Thread::*)(void) __ptr64,Il
blender.exe         :0x00007FF61D24CAC0  std::_Pad::_Call_func
ucrtbase.dll        :0x00007FFEA56B1B20  configthreadlocale
KERNEL32.DLL        :0x00007FFEA71C7020  BaseThreadInitThunk
ntdll.dll           :0x00007FFEA8342630  RtlUserThreadStart

Thread : 00002cc0
ntdll.dll           :0x00007FFEA838CDE0  NtWaitForSingleObject
KERNELBASE.dll      :0x00007FFEA51819D0  WaitForSingleObjectEx
blender.exe         :0x00007FF61E70E410  IlmThread_2_5::Semaphore::wait
blender.exe         :0x00007FF61E70DBE0  IlmThread_2_5::ThreadPool::numThreads
blender.exe         :0x00007FF61E70E880  std::_LaunchPad<std::unique_ptr<std::tuple<void (__cdecl IlmThread_2_5::Thread::*)(void) __ptr64,Il
blender.exe         :0x00007FF61D24CAC0  std::_Pad::_Call_func
ucrtbase.dll        :0x00007FFEA56B1B20  configthreadlocale
KERNEL32.DLL        :0x00007FFEA71C7020  BaseThreadInitThunk
ntdll.dll           :0x00007FFEA8342630  RtlUserThreadStart

Thread : 00003404
ntdll.dll           :0x00007FFEA838D8B0  ZwWaitForMultipleObjects
KERNELBASE.dll      :0x00007FFEA51ACA30  WaitForMultipleObjectsEx
combase.dll         :0x00007FFEA6329D70  RoGetServerActivatableClasses
combase.dll         :0x00007FFEA6329D70  RoGetServerActivatableClasses
combase.dll         :0x00007FFEA6329D70  RoGetServerActivatableClasses
combase.dll         :0x00007FFEA6329D70  RoGetServerActivatableClasses
KERNEL32.DLL        :0x00007FFEA71C7020  BaseThreadInitThunk
ntdll.dll           :0x00007FFEA8342630  RtlUserThreadStart

Thread : 00000df4
ntdll.dll           :0x00007FFEA83907B0  ZwWaitForWorkViaWorkerFactory
ntdll.dll           :0x00007FFEA8342680  TpReleaseCleanupGroupMembers
KERNEL32.DLL        :0x00007FFEA71C7020  BaseThreadInitThunk
ntdll.dll           :0x00007FFEA8342630  RtlUserThreadStart

Thread : 00002564
ntdll.dll           :0x00007FFEA838CDE0  NtWaitForSingleObject
KERNELBASE.dll      :0x00007FFEA51819D0  WaitForSingleObjectEx
nvoglv64.DLL        :0x00000000516D35A0  DrvValidateVersion
nvoglv64.DLL        :0x00000000514944EE  Symbols not available

Thread : 00000ab8
ntdll.dll           :0x00007FFEA838D8B0  ZwWaitForMultipleObjects
KERNELBASE.dll      :0x00007FFEA51ACA30  WaitForMultipleObjectsEx
KERNELBASE.dll      :0x00007FFEA51ACA10  WaitForMultipleObjects
nvoglv64.DLL        :0x0000000051706260  DrvPresentBuffers
nvoglv64.DLL        :0x0000000051706260  DrvPresentBuffers
nvoglv64.DLL        :0x0000000051A4EAA0  vk_optimusGetInstanceProcAddr
KERNEL32.DLL        :0x00007FFEA71C7020  BaseThreadInitThunk
ntdll.dll           :0x00007FFEA8342630  RtlUserThreadStart

Thread : 00002568
ntdll.dll           :0x00007FFEA838D8B0  ZwWaitForMultipleObjects
KERNELBASE.dll      :0x00007FFEA51ACA30  WaitForMultipleObjectsEx
KERNELBASE.dll      :0x00007FFEA51ACA10  WaitForMultipleObjects
nvoglv64.DLL        :0x0000000051706260  DrvPresentBuffers
KERNEL32.DLL        :0x00007FFEA71C7020  BaseThreadInitThunk
ntdll.dll           :0x00007FFEA8342630  RtlUserThreadStart

Thread : 00003650
win32u.dll          :0x00007FFEA550A0F0  NtUserMsgWaitForMultipleObjectsEx
USER32.dll          :0x00007FFEA6610AC0  MsgWaitForMultipleObjectsEx
nvoglv64.DLL        :0x00000000516D35A0  DrvValidateVersion
KERNEL32.DLL        :0x00007FFEA71C7020  BaseThreadInitThunk
ntdll.dll           :0x00007FFEA8342630  RtlUserThreadStart

Thread : 00002cbc
ntdll.dll           :0x00007FFEA838CDE0  NtWaitForSingleObject
KERNELBASE.dll      :0x00007FFEA51819D0  WaitForSingleObjectEx
tbb.dll             :0x00007FFE646336F0  tbb::thread_bound_filter::try_process_item
tbb.dll             :0x00007FFE646336F0  tbb::thread_bound_filter::try_process_item
ucrtbase.dll        :0x00007FFEA56B1B20  configthreadlocale
KERNEL32.DLL        :0x00007FFEA71C7020  BaseThreadInitThunk
ntdll.dll           :0x00007FFEA8342630  RtlUserThreadStart

Thread : 00002b20
ntdll.dll           :0x00007FFEA838CDE0  NtWaitForSingleObject
KERNELBASE.dll      :0x00007FFEA51819D0  WaitForSingleObjectEx
tbb.dll             :0x00007FFE646336F0  tbb::thread_bound_filter::try_process_item
tbb.dll             :0x00007FFE646336F0  tbb::thread_bound_filter::try_process_item
ucrtbase.dll        :0x00007FFEA56B1B20  configthreadlocale
KERNEL32.DLL        :0x00007FFEA71C7020  BaseThreadInitThunk
ntdll.dll           :0x00007FFEA8342630  RtlUserThreadStart

Thread : 000035c4
ntdll.dll           :0x00007FFEA838CDE0  NtWaitForSingleObject
KERNELBASE.dll      :0x00007FFEA51819D0  WaitForSingleObjectEx
tbb.dll             :0x00007FFE646336F0  tbb::thread_bound_filter::try_process_item
tbb.dll             :0x00007FFE646336F0  tbb::thread_bound_filter::try_process_item
ucrtbase.dll        :0x00007FFEA56B1B20  configthreadlocale
KERNEL32.DLL        :0x00007FFEA71C7020  BaseThreadInitThunk
ntdll.dll           :0x00007FFEA8342630  RtlUserThreadStart

Thread : 0000351c
ntdll.dll           :0x00007FFEA838CDE0  NtWaitForSingleObject
KERNELBASE.dll      :0x00007FFEA51819D0  WaitForSingleObjectEx
tbb.dll             :0x00007FFE646336F0  tbb::thread_bound_filter::try_process_item
tbb.dll             :0x00007FFE646336F0  tbb::thread_bound_filter::try_process_item
ucrtbase.dll        :0x00007FFEA56B1B20  configthreadlocale
KERNEL32.DLL        :0x00007FFEA71C7020  BaseThreadInitThunk
ntdll.dll           :0x00007FFEA8342630  RtlUserThreadStart

Thread : 000034b0
ntdll.dll           :0x00007FFEA838CDE0  NtWaitForSingleObject
KERNELBASE.dll      :0x00007FFEA51819D0  WaitForSingleObjectEx
tbb.dll             :0x00007FFE646336F0  tbb::thread_bound_filter::try_process_item
tbb.dll             :0x00007FFE646336F0  tbb::thread_bound_filter::try_process_item
ucrtbase.dll        :0x00007FFEA56B1B20  configthreadlocale
KERNEL32.DLL        :0x00007FFEA71C7020  BaseThreadInitThunk
ntdll.dll           :0x00007FFEA8342630  RtlUserThreadStart

Thread : 00001470
ntdll.dll           :0x00007FFEA838CDE0  NtWaitForSingleObject
KERNELBASE.dll      :0x00007FFEA51819D0  WaitForSingleObjectEx
tbb.dll             :0x00007FFE646336F0  tbb::thread_bound_filter::try_process_item
tbb.dll             :0x00007FFE646336F0  tbb::thread_bound_filter::try_process_item
ucrtbase.dll        :0x00007FFEA56B1B20  configthreadlocale
KERNEL32.DLL        :0x00007FFEA71C7020  BaseThreadInitThunk
ntdll.dll           :0x00007FFEA8342630  RtlUserThreadStart

Thread : 00002294
ntdll.dll           :0x00007FFEA838CDE0  NtWaitForSingleObject
KERNELBASE.dll      :0x00007FFEA51819D0  WaitForSingleObjectEx
tbb.dll             :0x00007FFE646336F0  tbb::thread_bound_filter::try_process_item
tbb.dll             :0x00007FFE646336F0  tbb::thread_bound_filter::try_process_item
ucrtbase.dll        :0x00007FFEA56B1B20  configthreadlocale
KERNEL32.DLL        :0x00007FFEA71C7020  BaseThreadInitThunk
ntdll.dll           :0x00007FFEA8342630  RtlUserThreadStart

Loaded Modules :
0x00007FF61A220000 2.9.3.0              blender.exe C:\Program Files\Blender Foundation\Blender 2.93\blender.pdb 
0x00007FFEA82F0000 10.0.19041.1288      ntdll.dll  
0x00007FFE8EDA0000 21.11.6809.0         aswhook.dll  
0x00007FFEA71B0000 10.0.19041.1348      KERNEL32.DLL  
0x00007FFEA5160000 10.0.19041.1387      KERNELBASE.dll  
0x00007FFEA65F0000 10.0.19041.1202      USER32.dll  
0x00007FFEA5500000 10.0.19041.1387      win32u.dll  
0x00007FFEA7310000 10.0.19041.1202      GDI32.dll  
0x00007FFEA5580000 10.0.19041.1387      gdi32full.dll  
0x00007FFEA5460000 10.0.19041.789       msvcp_win.dll  
0x00007FFEA5690000 10.0.19041.789       ucrtbase.dll  
0x00007FFEA7B60000 10.0.19041.1387      SHELL32.dll  
0x00007FFEA5530000 10.0.19041.1151      CFGMGR32.dll  
0x00007FFE61320000 10.0.19041.1081      OPENGL32.dll  
0x00007FFEA76F0000 7.0.19041.546        msvcrt.dll  
0x00007FFEA5C20000 10.0.19041.546       WS2_32.dll  
0x00007FFEA5B70000 10.0.19041.1052      ADVAPI32.dll  
0x00007FFEA5E40000 10.0.19041.1288      RPCRT4.dll  
0x00007FFEA7270000 10.0.19041.1415      sechost.dll  
0x00007FFEA6260000 10.0.19041.1348      combase.dll  
0x00007FFEA6130000 10.0.19041.1202      ole32.dll  
0x00007FFE64620000 2020.2.0.0           tbb.dll  
0x00007FFEA5F70000 10.0.19041.546       PSAPI.DLL  
0x00007FFEA59A0000 10.0.19041.1023      SHLWAPI.dll  
0x00007FFEA65C0000 10.0.19041.546       IMM32.dll  
0x00007FFE592E0000 8.0.1.0              openvdb.dll  
0x00007FFE32370000 3.9.2150.1013        python39.dll  
0x00007FFE2E210000 58.54.100.0          avcodec-58.dll  
0x00007FFE34580000 58.29.100.0          avformat-58.dll  
0x00007FFE8F050000 58.8.100.0           avdevice-58.dll  
0x00007FFEA5AA0000 10.0.19041.985       OLEAUT32.dll  
0x00007FFE611F0000 56.31.100.0          avutil-56.dll  
0x00007FFEA5130000 10.0.19041.1023      bcrypt.dll  
0x00007FFE630B0000 5.5.100.0            swscale-5.dll  
0x0000000070440000 1.0.28.0             libsndfile-1.dll  
0x00007FFE39DD0000 1.20.1.0             OpenAL32.dll  
0x00007FFE59250000 14.28.29910.0        MSVCP140.dll  
0x00007FFE9C2E0000 14.28.29910.0        VCRUNTIME140.dll  
0x00007FFE39CA0000 2.0.12.0             SDL2.dll  
0x00007FFE980B0000 14.28.29910.0        VCRUNTIME140_1.dll  
0x00007FFEA6C80000 10.0.19041.1237      SETUPAPI.dll  
0x00007FFE78120000 10.0.19041.1         AVIFIL32.dll  
0x000000006ACC0000                      libgmp-10.dll  
0x00007FFE9A730000 6.10.19041.1110      COMCTL32.dll  
0x00007FFE980A0000 2020.2.0.0           tbbmalloc_proxy.dll  
0x00007FFE9F800000 10.0.19041.546       VERSION.dll  
0x00007FFE6E9A0000 10.0.19041.1081      GLU32.dll  
0x00007FFE9E200000 10.0.19041.867       dbghelp.dll  
0x00007FFE94AE0000 10.0.19041.546       Secur32.dll  
0x00007FFE64600000 10.0.19041.1         AVICAP32.dll  
0x00007FFE9CE60000 10.0.19041.546       WINMM.dll  
0x00007FFE645D0000 10.0.19041.1         MSVFW32.dll  
0x00007FFE925F0000 10.0.19041.1         MSACM32.dll  
0x00007FFE644E0000 3.5.100.0            swresample-3.dll  
0x0000000070680000                      libfftw3-3.dll  
0x00007FFE9AC20000 10.0.19041.1         winmmbase.dll  
0x00007FFEA4F10000 10.0.19041.1266      SSPICLI.DLL  
0x00007FFE642A0000 2020.2.0.0           tbbmalloc.dll  
0x00007FFEA7100000 10.0.19041.1387      shcore.dll  
0x00007FFEA2F40000 10.0.19041.546       kernel.appcore.dll  
0x00007FFEA3140000 10.0.19041.1387      windows.storage.dll  
0x00007FFEA49A0000 10.0.19041.1320      Wldp.dll  
0x00007FFEA2A60000 10.0.19041.1266      uxtheme.dll  
0x00007FFEA5900000 10.0.19041.1415      bcryptPrimitives.dll  
0x00007FFEA4F60000 10.0.19041.844       profapi.dll  
0x00007FFEA7360000 2001.12.10941.16384  clbcatq.dll  
0x00007FFE9DDF0000 10.0.19041.1023      MMDevApi.dll  
0x00007FFEA4CF0000 10.0.19041.1151      DEVOBJ.dll  
0x00007FFE9B6F0000 10.0.19041.1023      AUDIOSES.DLL  
0x00007FFEA4E90000 10.0.19041.546       powrprof.dll  
0x00007FFEA4E70000                      UMPDC.dll  
0x00007FFEA7A40000 10.0.19041.1387      MSCTF.dll  
0x00007FFE8EFB0000 10.0.19041.1387      dataexchange.dll  
0x00007FFEA1300000 10.0.19041.1202      d3d11.dll  
0x00007FFEA1B30000 10.0.19041.1387      dcomp.dll  
0x00007FFEA3A40000 10.0.19041.1266      dxgi.dll  
0x00007FFE9F080000 10.0.19041.964       twinapi.appcore.dll  
0x00007FFE9F830000 10.0.19041.1288      AppXDeploymentClient.dll  
0x0000000050A80000 23.21.13.8813        nvoglv64.DLL  
0x00007FFE9F700000 10.0.19041.546       WTSAPI32.dll  
0x00007FFEA50D0000 10.0.19041.1266      WINTRUST.dll  
0x000001E5C3EF0000 10.0.19041.1320      CRYPT32.dll  
0x00007FFEA4B30000 10.0.19041.546       MSASN1.dll  
0x00007FFEA3D20000 10.0.19041.546       ntmarta.dll  
0x00007FFEA2C50000 10.0.19041.746       dwmapi.dll  
0x00007FFE98D00000 10.0.19041.546       dxcore.dll  
0x00007FFEA4D60000 10.0.19041.546       WINSTA.dll  
0x00007FFE8EC40000 10.0.19041.1387      textinputframework.dll  
0x00007FFEA26A0000 10.0.19041.746       CoreMessaging.dll  
0x00007FFEA2310000 10.0.19041.546       CoreUIComponents.dll  
0x00007FFEA20B0000 10.0.19041.1348      wintypes.dll  
0x00007FFE80A70000 10.0.19041.1023      explorerframe.dll  
0x00007FFEA0C00000 7.0.19041.1023       propsys.dll  
0x00007FFEA4900000 10.0.19041.546       CRYPTSP.dll  
0x00007FFEA4090000 10.0.19041.1052      rsaenh.dll  
0x00007FFEA48F0000 10.0.19041.546       CRYPTBASE.dll  
0x00007FFE857E0000 3.9.2150.1013        python3.DLL  
0x00007FFE63090000 3.9.2150.1013        _bz2.pyd  
0x00007FFE611C0000 3.9.2150.1013        _lzma.pyd  
0x00007FFE611A0000 3.9.2150.1013        _socket.pyd  
0x00007FFEA4400000 10.0.19041.546       IPHLPAPI.DLL  
0x00007FFE85600000 3.9.2150.1013        select.pyd  
0x00007FFE844B0000 3.9.2150.1013        _queue.pyd  
0x00007FFE39C70000 3.9.2150.1013        _ssl.pyd  
0x00007FFE34D30000 1.1.1.9              libssl-1_1.dll  
0x00007FFE32900000 1.1.1.9              libcrypto-1_1.dll  
0x00007FFE320A0000                      _multiarray_umath.cp39-win_amd64.pyd  
0x00007FFE5E180000                      _multiarray_tests.cp39-win_amd64.pyd  
0x00007FFE34D00000 3.9.2150.1013        _ctypes.pyd  
0x00007FFE83E00000                      libffi-7.dll  
0x00007FFE34C70000                      lapack_lite.cp39-win_amd64.pyd  
0x00007FFE31E70000                      _umath_linalg.cp39-win_amd64.pyd  
0x00007FFE34C20000 3.9.2150.1013        _decimal.pyd  
0x00007FFE59230000                      _pocketfft_internal.cp39-win_amd64.pyd  
0x00007FFE34B90000                      mtrand.cp39-win_amd64.pyd  
0x00007FFE34B60000                      bit_generator.cp39-win_amd64.pyd  
0x00007FFE34540000                      _common.cp39-win_amd64.pyd  
0x00007FFE34520000 3.9.2150.1013        _hashlib.pyd  
0x00007FFE31E30000                      _bounded_integers.cp39-win_amd64.pyd  
0x00007FFE34500000                      _mt19937.cp39-win_amd64.pyd  
0x00007FFE31E10000                      _philox.cp39-win_amd64.pyd  
0x00007FFE31DF0000                      _pcg64.cp39-win_amd64.pyd  
0x00007FFE31DD0000                      _sfc64.cp39-win_amd64.pyd  
0x00007FFE31D20000                      _generator.cp39-win_amd64.pyd  
0x00007FFE31CE0000 3.9.2150.1013        _elementtree.pyd  
0x00007FFE31CA0000 3.9.2150.1013        pyexpat.pyd  
0x00007FFE31C80000 3.9.2150.1013        _sqlite3.pyd  
0x00007FFE31B30000 3.34.0.0             sqlite3.dll  
0x00007FFEA7500000 11.0.19041.1320      urlmon.dll  
0x00007FFEA7790000 11.0.19041.1266      iertutil.dll  
0x00007FFEA5430000 10.0.19041.546       srvcli.dll  
0x00007FFEA58F0000 10.0.19041.546       netutils.dll  
0x00007FFE9AB40000 10.0.19041.746       Windows.Shell.ServiceHostBuilder.dll  
0x00007FFE99B10000 10.0.19041.1387      OneCoreUAPCommonProxyStub.dll  
0x00007FFEA4EE0000 10.0.19041.572       USERENV.dll  
0x00007FFE86640000 10.0.19041.844       Windows.StateRepositoryPS.dll  
0x00007FFE7FCF0000 10.0.19041.546       edputil.dll  
0x00007FFE8C590000 10.0.19041.746       MLANG.dll  
0x00007FFEA67A0000 11.0.19041.1320      WININET.dll  
0x00007FFE85840000 10.0.19041.1151      Windows.UI.AppDefaults.dll  
0x00007FFEA0730000 10.0.19041.1387      policymanager.dll  
0x00007FFEA4120000 10.0.19041.546       msvcp110_win.dll  
0x00007FFE85A70000 10.0.19041.1387      appresolver.dll  
0x00007FFE8E950000 10.0.19041.1266      Bcp47Langs.dll  
0x00007FFEA3BB0000 10.0.19041.546       SLC.dll  
0x00007FFEA3B80000 10.0.19041.546       sppc.dll  
0x00007FFE96F30000 10.0.19041.1081      OneCoreCommonProxyStub.dll  
0x00007FFE2D4C0000 23.21.13.8813        nvcuda.dll  
0x00007FFE31A10000 23.21.13.8813        nvfatbinaryLoader.dll  
0x00007FFE9BB80000 23.21.13.8813        nvapi64.dll  
0x00007FFE925C0000 2.1.0.0              OpenCL.dll  
0x00007FFE2B020000 23.21.13.8813        nvopencl.dll  

**Platform information:**
 - OIIO branch/version: 3.4.0
 - OS: Windows 10
 - C++ compiler:
 - Any non-default build flags when you build OIIO:

**IF YOU ALREADY HAVE A CODE FIX:** There is no need to file a separate issue,
please just go straight to making a [pull request](https://github.com/OpenImageIO/oiio/pulls).

**Comment by jessey-git** [comment]
Hmm, let me clarify a bit.  The bug here was first reported on the Blender bug tracker and I asked for it to be reported here since this is a problem within OIIO itself.  The version of OIIO used is 2.1 but it also repros using 2.2.  We don't have a too-easy way of checking 2.3 just yet.

```
blender.exe :0x00007FF61D41A550 OpenImageIO_v2_1::TIFFInput::palette_to_rgb
blender.exe :0x00007FF61D41A650 OpenImageIO_v2_1::TIFFInput::read_native_scanline
blender.exe :0x00007FF61D395F20 OpenImageIO_v2_1::ImageInput::read_native_scanlines
blender.exe :0x00007FF61D41B330 OpenImageIO_v2_1::TIFFInput::read_native_scanlines
blender.exe :0x00007FF61D395BE0 OpenImageIO_v2_1::ImageInput::read_native_scanlines
blender.exe :0x00007FF61D396D20 OpenImageIO_v2_1::ImageInput::read_scanlines
blender.exe :0x00007FF61D41D280 OpenImageIO_v2_1::TIFFInput::read_scanlines
blender.exe :0x00007FF61D395240 OpenImageIO_v2_1::ImageInput::read_image
blender.exe :0x00007FF61D395800 OpenImageIO_v2_1::ImageInput::read_image
blender.exe :0x00007FF61D395940 OpenImageIO_v2_1::ImageInput::read_image
```

In any case, here is the zip for the offending file (262k):
[Guerrero_r15m_Small.zip](https://github.com/OpenImageIO/oiio/files/7799988/Guerrero_r15m_Small.zip)

**Comment by lgritz** [comment]
Thanks, I seem to be able to reproduce (even in master) with that example file. Looking into it now, stay tuned...

**Comment by lgritz** [comment]
OK, I know now what's odd about the file. It's a palette image, but with a 16 bit palette. I guess I'd never run into a palette image that was >8 bytes per pixel. I can see exactly why this causes a crash now, let me figure out how hard it is to fix.

Interestingly, Mac Preview.app also refuses to open this file.

**Comment by lgritz** [comment]
Proposed fix in #3260
If that seems ok, I can backport it and cut a 2.2 patch for you.

**Comment by EvanAW** [comment]
> Proposed fix in #3260
> If that seems ok, I can backport it and cut a 2.2 patch for you.

Tagging @LazyDodo and @lichtwerk to loop them in as they are the developers in charge of Blender’s 3rd-party libraries and Long-term Support coordination, respectively. 

They can best answer if #3260 plus a patch will work for Blender.

**Comment by EvanAW** [comment]
> OK, I know now what's odd about the file. It's a palette image, but with a 16 bit palette. I guess I'd never run into a palette image that was >8 bits per pixel.

I think I know why.  I just examined the metadata with ExifTool.  This is a GeoTIFF file made with GRASS GIS 7.8.6 with GDAL 3.4.0

**Comment by LazyDodo** [comment]
> They can best answer if #3260 plus a patch will work for Blender.

I can confirm bumping oiio to v2.3.11.0+#3260 no longer crashes blender, if we want to backport this fix to the LTS releases is still under discussion on the blender side of things, for now if this fix were to land only in oiio 2.3.x that wouldn't be a problem.

**Comment by lgritz** [comment]
OK, it will definitely be in the next 2.3 patch release. If you need an official 2.2 with this also, let me know.
